runtime/compiler/optimizer/VectorAPIExpansion.cpp

```diff
@@ -1549,11 +1549,21 @@ TR_VectorAPIExpansion::unboxNode(TR::Node *parentNode, TR::Node *operand, vapiOb
    bool parentScalarized;
    bool parentVectorizedOrScalarized = isVectorizedOrScalarizedNode(parentNode, elementType, bitsLength,
                                                                     parentType, parentScalarized);
+   int32_t elementSize = OMR::DataType::getSize(elementType);
+   int32_t numLanes = bitsLength/8/elementSize;
+   TR::VectorLength vectorLength = OMR::DataType::bitsToVectorLength(bitsLength);
+   TR::ILOpCodes maskLoadOpCode;
+   TR::ILOpCodes maskConv;
 
    if ((operandObjectType != Vector && operandObjectType != Mask) ||
-        elementType != TR::Int8 ||
-        bitsLength != 128 ||
-        parentScalarized)  // TODO: support unboxing into scalars
+       elementType != TR::Int8 ||   // TODO: support more types (needs support in createClassesForBoxing)
+       bitsLength != 128 ||
+       parentScalarized  ||  // TODO: support unboxing into scalars
+       (operandObjectType == Mask &&
+        !isOpCodeImplemented(comp(),
+                             (maskConv = getLoadToMaskConversion(numLanes,
```

Comment:
I think if the load is a vector, you need to check support for the vload. I did find this expression hard to follow. For instance, you assign maskLoadOpCode inside the function call `getLoadToMaskConversion`.

Reply:
Sure, I will simplify it.

Reply:
We don't check for `vload` in several other places. I guess we assume that if a platform supports vectorization (check at the beginning of this opt) then `vload/vstore` are supported.

Reply:
Simplified the code.